File: Level 2/두 큐 합 같게 만들기/solution.py
from collections import deque


def solution(queue1, queue2):
    answer = 0

    q1 = deque(queue1)
    q2 = deque(queue2)

    sum1 = sum(queue1)
    sum2 = sum(queue2)

    while True:
        if answer == len(queue1) * 4:
            return -1

        if sum1 > sum2:
            num = q1.popleft()
            q2.append(num)
            sum1 -= num
            sum2 += num
            answer += 1
        elif sum1 < sum2:
            num = q2.popleft()
            q1.append(num)
            sum2 -= num
            sum1 += num
            answer += 1
        else:
            return answer

File: Level 2/두 큐 합 같게 만들기/solution.js
function solution(queue1, queue2) {
  let answer = 0;

  const max = queue1.length * 2;

  let sum1 = queue1.reduce((acc, cur) => acc + cur, 0);
  let sum2 = queue2.reduce((acc, cur) => acc + cur, 0);

  let [i, j] = [0, 0];

  while (true) {
    if (answer === max * 2) {
      return -1;
    }

    if (sum1 > sum2) {
      const value = queue1[i++];
      sum1 -= value;
      sum2 += value;
      queue2.push(value);
      answer++;
    } else if (sum1 < sum2) {
      const value = queue2[j++];
      sum2 -= value;
      sum1 += value;
      queue1.push(value);
      answer++;
    } else {
      return answer;
    }
  }
}
